src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -277,17 +300,20 @@ static boolean isAllowedSemiStructuredType(Object o) {
    * @return JSON array representing the input
    */
   static String validateAndParseArray(String columnName, Object input, long insertRowIndex) {
-    JsonNode jsonNode =
-        validateAndParseSemiStructuredAsJsonTree(columnName, input, "ARRAY", insertRowIndex);
-
-    // Non-array values are ingested as single-element arrays, mimicking the Worksheets behavior
-    if (!jsonNode.isArray()) {
-      jsonNode = objectMapper.createArrayNode().add(jsonNode);
+    String result = validateAndParseSemiStructured(columnName, input, "ARRAY", insertRowIndex);
+    if (result.isEmpty()) {
+      // Empty input is ingested as an array of null
+      result =
+          JsonToken.START_ARRAY.asString()
+              + JsonToken.VALUE_NULL.asString()
+              + JsonToken.END_ARRAY.asString();
```

Comment:
Does this mean an  `"arrayKey": NULL` and/or `"arrayKey": ""` will get converted into `"arrayKey": []` ?

The latter needs to throw, and the former should stay as null (since NULL != empty array)

Reply:
This is for the `ARRAY` data type, and it means that `""` gets converted into `[null]`. It is an inconsistency with other semi-structured data types, and we should address - I created SNOW-1641130.

Reply:
Is this a new behavior or not?

Reply:
It is not a new behavior, see [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java#L284) for equivalent behavior in existing code. It should be addressed as a separate issue.

Reply:
sg if it's not a new behavior, thanks